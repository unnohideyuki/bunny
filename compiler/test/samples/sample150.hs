qsort :: Ord a => [a] -> [a]
qsort []     = []
qsort (x:xs) = qsort smaller ++ [x] ++ qsort larger
               where
                 smaller = [a | a <- xs, a <= x]
                 larger  = [b | b <- xs, b > x]


myShowList xs = foldr (\x s -> show x ++ " " ++ s) "" xs

main :: IO ()
main =
  do let helo = "Hello, World!"
     -- putStrLn helo
     putStrLn.myShowList $ qsort [3, 1, 4, 1, 5, 9, 2, 6, 5]
     putStrLn $ myShowList $ qsort helo
