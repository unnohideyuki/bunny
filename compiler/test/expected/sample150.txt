1 1 2 3 4 5 5 6 9 
' ' '!' ',' 'H' 'W' 'd' 'e' 'l' 'l' 'l' 'o' 'o' 'r' 
